#!/usr/bin/env python

PACKAGE="fre_row_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       name,                type,   level, description,           default, min, max
gen.add("p_gain",            double_t, 0,  "p gain for turning: ", 5,    0.0, 100.0)
gen.add("i_gain",            double_t, 0,  "i gain for turning: ", 0.03,    0.0, 0.5)
gen.add("d_gain",            double_t, 0,  "d gain for turning: ", 0,    0.0, 0.5)

gen.add("use_speed_control", bool_t,   0, "use_speed_control: ",   True)
gen.add("speed_low",         double_t, 0, "minimal drive speed: ", 0.6,     0.0, 2.0)
gen.add("speed_high",        double_t, 0, "max drive speed: ",     2.2,     0.0, 5.0)

gen.add("end_line_frame_count_threshold", int_t, 0, "end_line_frame_count_threshold: ", 15, 1, 100)

gen.add("ray_length",        double_t, 0, "ray_length: ",          1.3,     0.0, 5.0)

gen.add("min_cone_width",    double_t, 0, "min cone width in deg: ", 15.0,  0.0, 90.0)
gen.add("max_cone_width",    double_t, 0, "max cone width in deg: ", 45.0,  0.0, 90.0)

gen.add("multi_cone_enabled", bool_t,  0, "if true, use three cones", True)

gen.add("obstacle_angle",    double_t, 0, "obstacle_angle in deg: ", 2.0,   0.0, 10.0)
gen.add("min_obs_density",   double_t, 0, "min_obs_density: ",     0.5,     0.0, 1.0)

exit(gen.generate(PACKAGE, "crawl_row_param", "CrawlRowParam"))
